### DISCUSSION TITLE
MITM Logging?

### DISCUSSION THREAD

**Discussion by kenberland** [eclipse-lsp4j/lsp4j#812] [General]
Is there an easy way to cause lsp4j to log the entire client-server conversation to a file? Perhaps an environment variable or addition to `logging.properties`?

**Comment by jonahgraham** [comment]
Not that I know of. There is [`TracingMessageConsumer`](https://github.com/eclipse-lsp4j/lsp4j/blob/88868d3506d317b12cf0bfd1e82a254489f8c47b/org.eclipse.lsp4j.jsonrpc/src/main/java/org/eclipse/lsp4j/jsonrpc/TracingMessageConsumer.java#L36) that logs in such a format that the now discontinued [Language Server Protocol Inspector](https://github.com/microsoft/language-server-protocol-inspector) understood. But that requires the LSP4J to be launched in the correct way (using [`Launcher.traceMessages`](https://github.com/eclipse-lsp4j/lsp4j/blob/88868d3506d317b12cf0bfd1e82a254489f8c47b/org.eclipse.lsp4j.jsonrpc/src/main/java/org/eclipse/lsp4j/jsonrpc/Launcher.java#L295)) which is normally handled by the next layer up.

**Comment by kenberland** [comment]
I seem to be getting something with this [hack](https://gist.github.com/kenberland/79c1badb0068fc0123abfd837a315e98).

**Comment by nixel2007** [comment]
We use PrintWriter for that:

https://github.com/1c-syntax/bsl-language-server/blob/develop/src/main/java/com/github/_1c_syntax/bsl/languageserver/cli/lsp/LanguageServerLauncherConfiguration.java

We use extended configurable version of PrintWriter: https://github.com/1c-syntax/bsl-language-server/blob/develop/src/main/java/com/github/_1c_syntax/bsl/languageserver/cli/lsp/FileAwarePrintWriter.java

But you can pass the generic one.
